plugin-server/src/sentry.ts

```diff
@@ -0,0 +1,60 @@
+import * as Sentry from '@sentry/node'
+import * as Tracing from '@sentry/tracing'
+import { Span, SpanContext, TransactionContext } from '@sentry/types'
+import { AsyncLocalStorage } from 'node:async_hooks'
+
+import { PluginsServerConfig } from './types'
+
+// Must require as `tsc` strips unused `import` statements and just requiring this seems to init some globals
+require('@sentry/tracing')
+
+const asyncLocalStorage = new AsyncLocalStorage<Tracing.Span>()
+
+// Code that runs on app start, in both the main and worker threads
+export function initSentry(config: PluginsServerConfig): void {
+    if (config.SENTRY_DSN) {
+        Sentry.init({
+            dsn: config.SENTRY_DSN,
+            normalizeDepth: 8, // Default: 3
+            initialScope: {
+                tags: {
+                    PLUGIN_SERVER_MODE: config.PLUGIN_SERVER_MODE,
+                },
+            },
+            tracesSampleRate: config.SENTRY_PLUGIN_SERVER_TRACING_SAMPLE_RATE,
```

Comment:
Note: for the Python app [we ended up using a sampler function](https://github.com/PostHog/posthog/pull/10866/files). Not necessary for this PR but keep it in mind for future iterations